hudi-examples/hudi-examples-spark-slim/src/main/java/org/apache/hudi/examples/quickstartslim/HoodieSparkQuickstartSlim.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.examples.quickstartslim;
+
+import org.apache.hudi.QuickstartUtils;
+import org.apache.hudi.common.model.HoodieAvroPayload;
+import org.apache.hudi.common.model.WriteOperationType;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.examples.commonslim.HoodieExampleDataGeneratorSlim;
+import org.apache.hudi.examples.commonslim.HoodieExampleSparkUtilsSlim;
+import org.apache.hudi.exception.HoodieException;
+import org.apache.hudi.keygen.constant.KeyGeneratorOptions;
+
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+
+import java.util.List;
+
+import static org.apache.hudi.config.HoodieWriteConfig.TBL_NAME;
+import static org.apache.spark.sql.SaveMode.Append;
+import static org.apache.spark.sql.SaveMode.Overwrite;
+
+public final class HoodieSparkQuickstartSlim {
+
+  private HoodieSparkQuickstartSlim() {
+  }
+
+  public static void main(String[] args) {
+    if (args.length < 2) {
+      System.err.println("Usage: HoodieWriteClientExample <tablePath> <tableName>");
+      System.exit(1);
+    }
+    String tablePath = args[0];
+    String tableName = args[1];
+
+    SparkSession spark = HoodieExampleSparkUtilsSlim.defaultSparkSession("Hudi Spark basic example");
+    try (JavaSparkContext jsc = new JavaSparkContext(spark.sparkContext())) {
+      runQuickstart(jsc, spark, tableName, tablePath);
+    }
+  }
+
+  /**
+   * Visible for testing
+   */
+  public static void runQuickstart(JavaSparkContext jsc, SparkSession spark, String tableName, String tablePath) {
```

Comment:
we have duplicated the code b/w hudi-examples-spark and hudi-examples-spark-slim. can you create a follow up jira on this. we should try to dedup later.